ember.md

```diff
@@ -0,0 +1,457 @@
+# Ember.js Style Guide
+
+## Table Of Contents
+
+* [General](#general)
+* [Organizing your modules](#organizing-your-modules)
+* [Models](#models)
+* [Controllers](#controllers)
+* [Templates](#templates)
+* [Routing](#routing)
+* [Ember data](#ember-data)
+
+## General
+
+### Import what you use, do not use globals
+
+For Ember Data, we should import `DS` from `ember-data`, and then
+destructure our desired modules.
+For Ember, use destructuring [as Ember's internal organization is
+not intuitive and difficult to grok, and we should wait until Ember has been
+correctly modularized.](https://github.com/ember-cli/ember-cli-shims/issues/53)
+
+Once Ember has officially adopted shims, we will prefer shims over
+destructuring.
+
+```javascript
+// Good
+
+import Ember from 'ember';
+import DS from 'ember-data';
+
+const {
+  computed,
+} = Ember;
+
+const {
+  attr,
+} = DS;
+
+export default DS.Model.extend({
+  firstName: attr('string'),
+  lastName: attr('string'),
+
+  fullName: computed('firstName', 'lastName', function() {
+    // Code
+  }),
+});
+
+// Bad
+
+export default DS.Model.extend({
+  firstName: DS.attr('string'),
+  lastName: DS.attr('string'),
+
+  fullName: Ember.computed('firstName', 'lastName', function _fullName() {
+    // Code
+  }),
+});
+```
+
+### Don't use Ember's prototype extensions
+
+Avoid Ember's `Date`, `Function` and `String` prototype extensions. Prefer the
+corresponding functions from the `Ember` object.
+
+```javascript
+// Good
+
+const {
+  String: { w },
+  computed,
+  on,
+} = Ember;
+
+export default Model.extend({
+  hobbies: w('surfing skateboarding skydiving'),
+  fullName: computed('firstName', 'lastName', function _fullName() { ... }),
+  didError: on('error', function _didError() { ... })
+});
+
+// Bad
+
+export default Model.extend({
+  hobbies: 'surfing skateboarding skydiving'.w(),
+  fullName: function() { ... }.property('firstName', 'lastName'),
+  didError: function() { ... }.on('error')
+});
+```
+
+### Use `get` and `set`
+
+Calling `someObj.get('prop')` couples your code to the fact that
+`someObj` is an Ember Object. It prevents you from passing in a
+POJO, which is sometimes preferable in testing. It also yields a more
+informative error when called with `null` or `undefined`.
```

Comment:
This is one thing that is pretty different from the current state of our codebase. I'd like to try it, for the reasons given here.

I changed the example a little bit though. I feel more strongly about using `get(someObj, 'prop')` than `get(this, 'prop')`